.github/workflows/ci.yml

```diff
@@ -21,19 +21,26 @@ jobs:
       if: runner.os == 'Linux' && matrix.ghc == '8.2'
       run: |
         sudo apt-get install libncurses5 libtinfo5
-    - uses: haskell/actions/setup@v2
-      id: setup-haskell-cabal
-      with:
-        ghc-version: ${{ matrix.ghc }}
+    - if: runner.os == 'Linux' # this is actually a bug in github images: https://github.com/actions/runner-images/pull/6770
+      name: Fix ghcup dir
+      run: echo GHCUP_INSTALL_BASE_PREFIX=/usr/local >> $GITHUB_ENV
+    - name: Setup toolchain
+      run: |
+        which ghcup
+        ghcup install cabal recommended
+        ghcup install ghc --set ${{ matrix.ghc }}
```

Comment:
Would be nice if `ghcup` could also install the prerequisites or print a command that could be executed to do so.

As it stands, you might have to continue adding steps to install prerequisites to various ghc versions --- but you might not be scared of that.

Reply:
> Would be nice if ghcup could also install the prerequisites

It will never do so.

Reply:
> As it stands, you might have to continue adding steps to install prerequisites to various ghc versions --- but you might not be scared of that.

Yes, I'm fine with that. The action doesn't add any value and instead makes it hard to reason about what exactly is installed and from what source.